src/authentication.ts

```diff
@@ -32,10 +32,22 @@ function isTokenResponse(body: TAzureADErrorResponse | TTokenResponse): body is
   return (body as TTokenResponse).access_token !== undefined
 }
 
-function postWithFormData(tokenEndpoint: string, formData: FormData): Promise<TTokenResponse> {
+function buildUrlEncodedRequest(tokenRequest: TTokenRequest): string {
+  let s = ''
+  for (const pair of Object.entries(tokenRequest)) {
+    s += (s ? '&' : '') + pair[0] + '=' + encodeURIComponent(pair[1])
+  }
+  return s
+}
```

Comment:
This probably works just fine. But I like axios way of url encoding x-forms using URLSearchParams better.
I think it's more readable, and probably handles some edge cases better.
See https://axios-http.com/docs/urlencoded for reference.

Reply:
I do like those methods, and I have used them. However in this case we have a very restricted set of parameters that will ever be encountered due to the nature of the spec. I prefer this simple method since it doesn't involve any additional libraries!

If it ever fails I'll help re-work it... deal?

Reply:
`URLSearchParams` is a builtin class. No additional library required.
I would still like to see this change, but accept your argument.
I'll leave it up to you :)